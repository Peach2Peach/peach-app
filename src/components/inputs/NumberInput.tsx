import { enforceDecimalsFormat } from '../../utils/format'
import { Input, InputProps } from './Input'

type Props = InputProps & {
  decimals?: number
}
export const NumberInput = ({ decimals = 0, onChange, onSubmit, ...props }: Props) => (
  <Input
    {...props}
    keyboardType="numeric"
    onChange={onChange ? (number: string) => onChange(enforceDecimalsFormat(number, decimals)) : undefined}
    onSubmit={onSubmit ? (number: string) => onSubmit(enforceDecimalsFormat(number, decimals)) : undefined}
  />
)
