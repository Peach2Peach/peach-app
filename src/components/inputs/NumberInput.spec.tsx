import { createRenderer } from 'react-test-renderer/shallow'
import { act, fireEvent, render } from 'test-utils'
import { NumberInput } from './NumberInput'

describe('NumberInput', () => {
  it('renders correctly', () => {
    const renderer = createRenderer()
    renderer.render(<NumberInput />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
  it('should enforce number format on change', () => {
    const onChangeMock = jest.fn()
    const { getByPlaceholderText } = render(
      <NumberInput decimals={2} placeholder="placeholder" onChangeText={onChangeMock} />,
    )
    const input = getByPlaceholderText('placeholder')
    act(() => {
      fireEvent.changeText(input, '1,523')
    })
    expect(onChangeMock).toHaveBeenLastCalledWith('1.52')
  })
})
