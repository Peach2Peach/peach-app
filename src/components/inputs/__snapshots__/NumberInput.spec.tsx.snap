// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput renders correctly 1`] = `
<Input
  keyboardType="numeric"
/>
`;
